lib/src/main/java/io/ably/lib/types/ClientOptions.java

```diff
@@ -31,6 +31,47 @@ public ClientOptions(String key) throws AblyException {
 	}
 
 	/**
+	 * Create a new copy of object
+	 * @param options source object for a copy
+	 * @return copied object
+	 */
+	public static ClientOptions copy(ClientOptions options) {
+		ClientOptions copy = new ClientOptions();
+		AuthOptions.copyAttributes(copy,options);
+		return ClientOptions.copyAttributes(copy, options);
+	}
+
+	/**
+	 * Copy the values of each attribute
+	 * @param src source object for copy attributes
+	 * @param dest destination object
+	 * @return source object with new attributes
+	 */
+	public static ClientOptions copyAttributes(ClientOptions src, ClientOptions dest) {
```

Comment:
What is this needed for and how does this relate to `RSA8e, RSA9h, RSA10j`?   I am not following how `ClientOptions` would ever be passed in again to the client library as you can only instance it once.